examples/ImageRecon/GEOmetrics/utils.py

```diff
@@ -14,489 +14,467 @@
-	splitting_connecting_side_3_2 = torch.ones(splitting_face_list_len).view(-1,1,1).long().to(device)*2
-	splitting_connecting_side_3_3 = splitting_face_list_values[:,2,1].view(-1,1,1)
-
-	splitting_face_number_3 = new_faces_indecies_3.view(-1,1,1)
-	new_splitting_face_list_3_1 = torch.cat((splitting_connecting_face_3_1,splitting_connecting_side_3_1, splitting_face_number_3 ), dim = 2 )
-	new_splitting_face_list_3_2 = torch.cat((splitting_connecting_face_3_2,splitting_connecting_side_3_2, splitting_face_number_3 ), dim = 2 )
-	new_splitting_face_list_3_3 = torch.cat((splitting_connecting_face_3_3,splitting_connecting_side_3_3, splitting_face_number_3 ), dim = 2 )
-	new_splitting_face_list_3 = torch.cat((new_splitting_face_list_3_1, new_splitting_face_list_3_2, new_splitting_face_list_3_3), dim = 1)
-
-	# conplete new face_list is made 
-
-	new_splitting_face_list = torch.cat((new_unsplitting_face_list, new_splitting_face_list_1, new_splitting_face_list_2, new_splitting_face_list_3))
-	split_faces = faces_verts[splitting_faces_indecies]
-
-	# now to make the new vertex 
-	vertex_count = meshes['update'][index].vertices.shape[0]
-	new_len = split_faces.shape[0] + vertex_count
-
-	# select the vertices of the faces to be split 
-	x_f = split_faces[:,0]
-	y_f = split_faces[:,1]
-	z_f = split_faces[:,2]
-	verts_and_features = torch.cat((meshes['update'][index].vertices, features), dim = 1) 
-	x_v = verts_and_features[x_f] 
-	y_v = verts_and_features[y_f]
-	z_v = verts_and_features[z_f]
-	#average the featurs and position 
-	v1 = x_v/3  + y_v/3 + z_v/3 
-	verts_and_features = torch.cat((verts_and_features, v1))
-	verts = verts_and_features[:,:3]
-
-	features = verts_and_features[:,3:]
-	v1_inds = (vertex_count + torch.arange(split_faces.shape[0])).to(device).view(-1,1)
-	x_f = x_f.view(-1,1)
-	y_f = y_f.view(-1,1)
-	z_f = z_f.view(-1,1)
-	# define verts of new faces 
-	new_face_1 = torch.cat((x_f ,y_f ,v1_inds) , dim =1)
-	new_face_2 = torch.cat((v1_inds , y_f, z_f ) , dim =1)
-	new_face_3 = torch.cat((x_f ,v1_inds,z_f) , dim =1)
-	# name new face_verts array by appending then to old in order previously defined 
-	face_archive = torch.cat( (faces_verts, new_face_1, new_face_2, new_face_3))
-
-	faces = face_archive[new_splitting_face_list[:,0,2]]
-
-	return verts, faces, face_archive, new_splitting_face_list, features
+    loss = kal.metrics.mesh.edge_length(meshes['update'][0])
+    loss += kal.metrics.mesh.edge_length(meshes['update'][1])
+    loss += kal.metrics.mesh.edge_length(meshes['update'][2])
+    return loss
 
 
+def loss_lap(meshes): 
+    loss = .3 * kal.metrics.mesh.laplacian_loss(meshes['init'][0], meshes['update'][0])
+    loss += kal.metrics.mesh.laplacian_loss(meshes['init'][1], meshes['update'][1])
+    loss += kal.metrics.mesh.laplacian_loss(meshes['init'][2], meshes['update'][2])
+
+    loss += torch.sum((meshes['init'][1].vertices - meshes['update'][1].vertices)**2, 1).mean() * .0666
```

Comment:
spaces missing around "**"

Reply:
It's pretty common to not put spaces around the power operator (because it's the highest priority operation) from the projects I've come across, but I've checked out Pytorch and they seem add spaces, so I will go ahead an change. Note that Flake8 does not flag this as a format error.